pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -763,24 +769,15 @@ func liveMigrationMonitor(vmi *v1.VirtualMachineInstance, l *LibvirtDomainManage
 	completionTimeoutPerGiB := options.CompletionTimeoutPerGiB
 
 	acceptableCompletionTime := completionTimeoutPerGiB * getVMIMigrationDataSize(vmi)
+
+	// Active DomainJobStats flag
+	var activeDomainJobStatsFlag = libvirt.DomainGetJobStatsFlags(0)
+
 monitorLoop:
 	for {
 
-		select {
-		case passedErr := <-migrationErr:
-			if passedErr != nil {
-				logger.Reason(passedErr).Error("Live migration failed")
-				var abortStatus v1.MigrationAbortStatus
-				if strings.Contains(passedErr.Error(), "canceled by client") {
-					abortStatus = v1.MigrationAbortSucceeded
-				}
-				l.setMigrationResult(vmi, true, fmt.Sprintf("Live migration failed %v", passedErr), abortStatus)
-				break monitorLoop
-			}
-		default:
-		}
-
-		stats, err := dom.GetJobInfo()
+		logger.V(4).Infof("Getting JobStats with flag %d", int(activeDomainJobStatsFlag))
+		stats, err := dom.GetJobStats(activeDomainJobStatsFlag)
```

Comment:
we all `break` if this fails, which i think sends us out of the monitor loop does it not?  is what we're doing here safe? 

Previously we'd monitor until the migrationErr was hit, and report the result then exit before this code was hit, so maybe this issue never occurred?

Reply:
`break` sends us to the beginning of the loop and not out of the monitor loop. `break monitorLoop` sends us out.
that should allow us to process the `migrationErr` still

Reply:
> break sends us to the beginning of the loop and not out of the monitor loop.

how? maybe i'm missing something obvious?

```
cat >> scratch.go < EOF
package main                            
                                   
import (                             
        "fmt"
)                                                
                                                  
func main() {                                                     
        iter := -1
                                                                                               
monitorLoop:                                                                                   
        for {   
                iter++
                if iter == 0 {
                        fmt.Println("break")
                        break                                                                  
                } else {
                        fmt.Println("break monitorloop")
                        break monitorLoop                                                      
                }
                iter++                                                                         
        }                                                                                      
                                                                                               
        fmt.Println("out of loop")
} 
EOF
```
$ go build scratch.go && ./scratch 
break
out of loop

Reply:
no, I'm obviously wrong. In my head that was inside a switch statement. long day...